test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetTrustTests.cs

```diff
@@ -0,0 +1,555 @@
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Trust_CertificateFingerPrint_WithExistingSigner_UpdatesItSuccess(bool allowUntrustedRoot)
+        {
+            // Arrange
+            var nugetConfigFileName = "NuGet.Config";
+            var package = new SimpleTestPackageContext();
+
+            using (SimpleTestPathContext pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            using (MemoryStream zipStream = await package.CreateAsStreamAsync())
+            using (TrustedTestCert<TestCertificate> trustedTestCert = SigningTestUtility.GenerateTrustedTestCertificate())
+            {
+                var certFingerprint = SignatureTestUtility.GetFingerprint(trustedTestCert.Source.Cert, HashAlgorithmName.SHA256);
+                var repoServiceIndex = "https://serviceindex.test/v3/index.json";
+                var signedPackagePath = await SignedArchiveTestUtility.RepositorySignPackageAsync(trustedTestCert.Source.Cert, package, pathContext.PackageSource, new Uri(repoServiceIndex));
+
+                var config = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <trustedSigners>
+        <author name=""MyCompanyCert"">
+            <certificate fingerprint=""abcdefg"" hashAlgorithm=""SHA256"" allowUntrustedRoot=""false"" />
+        </author>
+    </trustedSigners>
+</configuration>";
+
+                SettingsTestUtils.CreateConfigurationFile(nugetConfigFileName, pathContext.WorkingDirectory, config);
+                var nugetConfigPath = Path.Combine(pathContext.WorkingDirectory, nugetConfigFileName);
+                var allowUntrustedRootArg = allowUntrustedRoot ? "--allow-untrusted-root" : string.Empty;
+                var allowUntruestedRootValue = allowUntrustedRoot ? "true" : "false";
+                var authorName = "MyCompanyCert";
+
+                // Act
+                var resultAdd = _msbuildFixture.RunDotnet(
+                    pathContext.SolutionRoot,
+                    $"nuget trust certificate {authorName} {certFingerprint} {allowUntrustedRootArg}  --algorithm SHA256 --configfile {nugetConfigPath}");
+
+                // Assert
+                resultAdd.Success.Should().BeTrue();
+                resultAdd.AllOutput.Should().Contain(string.Format(CultureInfo.CurrentCulture, "Successfully updated the trusted signer '{0}'.", authorName));
+
+                string expectedResult = SettingsTestUtils.RemoveWhitespace($@"<?xml version=""1.0"" encoding=""utf-8""?>
+                <configuration>
+                    < trustedSigners>
+                        <author name = ""{authorName}"">
+                                < certificate fingerprint = ""abcdefg"" hashAlgorithm = ""SHA256"" allowUntrustedRoot = ""false""/>
+                                < certificate fingerprint = ""{certFingerprint}"" hashAlgorithm = ""SHA256"" allowUntrustedRoot = ""{allowUntruestedRootValue}""/>
```

Comment:
It's great that we could add to the existing trusted signers.
Do we have the same behavior in NuGet.exe now?
I don't think we have documented it. Perhaps adding this in doc?